File: app/src/main/java/com/jiujiu/mvvmTemplate/worker/PrePopulateDataWorker.java
package com.jiujiu.mvvmTemplate.worker;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import com.jiujiu.mvvmTemplate.data.DataManager;
import com.squareup.inject.assisted.Assisted;
import com.squareup.inject.assisted.AssistedInject;

public class PrePopulateDataWorker extends Worker {
    private static final String TAG = "PrePopulateDataWorker";

    private DataManager dataManager;

    public PrePopulateDataWorker(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @AssistedInject
    public PrePopulateDataWorker(@Assisted @NonNull Context context, @Assisted @NonNull WorkerParameters workerParams, DataManager dataManager) {
        super(context, workerParams);
        this.dataManager = dataManager;
    }

    @NonNull
    @Override
    public Result doWork() {
        dataManager.prePupulateData();
        return Result.success();
    }

    @AssistedInject.Factory
    public interface Factory extends CustomWorkerFactory {
    }
}

File: app/src/main/java/com/jiujiu/mvvmTemplate/worker/PrePopulateDataWorker.kt
package com.jiujiu.mvvmTemplate.worker

import android.content.Context
import androidx.work.CoroutineWorker
import androidx.work.WorkerParameters

import com.jiujiu.mvvmTemplate.data.DataManager
import com.squareup.inject.assisted.Assisted
import com.squareup.inject.assisted.AssistedInject
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.coroutineScope

class PrePopulateDataWorker @AssistedInject constructor(
        @Assisted context: Context,
        @Assisted workerParams: WorkerParameters,
        private val dataManager: DataManager
) : CoroutineWorker(context, workerParams) {

    override val coroutineContext = Dispatchers.IO

    override suspend fun doWork(): Result {
        return coroutineScope {
            dataManager.prePopulateData()
            Result.success()
        }
    }

    @AssistedInject.Factory
    interface Factory : CustomWorkerFactory
}
